### PR TITLE
[Feature: #162727575] Track Comment Edit History

### PR DESCRIPTION
**Pull request by ebenezerdon**
#### What does this PR do?
This PR adds a feature for editing, deleting, getting and tracking comments edit history.

#### Description of Task to be completed?
A user should be able to see comments and the previous versions of comments made on an article after previous edits.

#### How should this be manually tested?
- run `db:migrate`
- send a `PATCH` request to `/api/v1/articles/:slug/comments/:id`
- the comment should be successfully updated

- send a `GET` request to `/api/v1/articles/:slug/comments/:id/edits`
- you should get a response including the comment edit history

- after creating a comment, send a `DELETE` request to `/api/v1/articles/:slug/comments/:id`
- the comment should be deleted successfully

#### What are the relevant pivotal tracker stories?
[#162727575](https://www.pivotaltracker.com/story/show/162727575)

### CODE REVIEW

File: controllers/comment.js [hunk 1]
```diff
@@ -1,23 +1,60 @@
 import Sequelize from 'sequelize';
+import moment from 'moment';
```
Comment (oreoluwade) [inline controllers/comment.js]: There's a time-manipulation package we are using in the application already. The name is `date-fns`.

Shouldn't be hard to use. Check it out and just replace the instances of moment with the way `date-fns` implements theirs.

Comment (ebenezerdon) [reply controllers/comment.js]: I've made the change

File: controllers/comment.js [hunk 2]
```diff
@@ -1,23 +1,60 @@
...
+    await commentRow.update({ comment });
+    const updatedComment = Object.values(comment).slice(-1)[0];
+    return res.json({ updatedComment });
+  },
+
+  async getComment(req, res) {
+    const { id } = req.params;
+    const commentRow = await Comment.findOne({ where: { id } });
+    const { comment } = commentRow;
+    const latestEdit = Object.values(comment).slice(-1)[0];
```
Comment (oreoluwade) [inline controllers/comment.js:40]: What is the `slice`used for here?

Comment (ebenezerdon) [reply controllers/comment.js:40]: The `slice(-1)` returns an array of just the last value, and then `[0]` returns the value itself

**Comment by oreoluwade** [comment]
Excellent work!

**Comment by ebenezerdon** [comment]
Thanks!

**Pull request merged**